nmap.go

```diff
@@ -83,11 +82,14 @@ func (s *Scanner) Run() (*Run, error) {
 		return nil, ErrScanTimeout
 	case <-done:
 		// Scan finished before timeout.
+		var nmapErrors []NmapErrors
 		if stderr.Len() > 0 {
-			return nil, errors.New(strings.Trim(stderr.String(), ".\n"))
+			for _, v := range RemoveDuplicatesFromSlice(strings.Split(strings.Trim(stderr.String(), "\n"), "\n")) {
```

Comment:
Please add a comment above this line to explain it, since it contains multiple nested calls it might be confusing to someone who stumbles upon it what the purpose of the line is :)

Reply:
What do you think about something like this?

```suggestion
			// List all unique errors returned by nmap.
			for _, v := range RemoveDuplicatesFromSlice(strings.Split(strings.Trim(stderr.String(), "\n"), "\n")) {
```